external/fv3fit/fv3fit/data/tfdataset.py

```diff
@@ -56,6 +62,45 @@ def get_n_windows(n_times: int, window_size: int) -> int:
     return (n_times - 1) // (window_size - 1)
 
 
+@register_tfdataset_loader
+@dataclasses.dataclass
+class CycleGANLoader(TFDatasetLoader):
+
+    domain_configs: List[TFDatasetLoader] = dataclasses.field(default_factory=list)
+    batch_size: int = 1
+
+    def open_tfdataset(
+        self, local_download_path: Optional[str], variable_names: Sequence[str],
+    ) -> tf.data.Dataset:
+        datasets = []
+        for config in self.domain_configs:
+            datasets.append(config.open_tfdataset(local_download_path, variable_names))
+        return tf.data.Dataset.zip(tuple(datasets))
+
+    @classmethod
+    def from_dict(cls, d: dict) -> "CycleGANLoader":
+        with prevent_recursion():
+            domain_configs = [
+                tfdataset_loader_from_dict(domain_config)
+                for domain_config in d["domain_configs"]
+            ]
+        return CycleGANLoader(domain_configs=domain_configs)
+
+
+RECURSING = False
+
+
+@contextlib.contextmanager
+def prevent_recursion():
```

Comment:
Needed so when checking if something is a CycleGANLoader we don't check if it's a CycleGANLoader and in order to do so, check that it's a CycleGANLoader, etc.

Reply:
OK so it turns out I don't actually need this, so I deleted it.

It turns out we don't check that the config itself is a CycleGANLoader to check that it's a CycleGANLoader, we check whether it's sub-members are CycleGANLoaders. Eventually we'll run out of sub-members so this can never be an infinite loop.